lib/segment/benches/sparse_index_search.rs

```diff
@@ -82,17 +119,19 @@ fn sparse_vector_index_search_benchmark(c: &mut Criterion) {
             &stopped,
         )
         .unwrap();
+    let pb = progress("Indexing (3/3)", vectors_len);
     sparse_vector_index_mmap
-        .build_index(permit, &stopped)
+        .build_index_with_progress(permit, &stopped, || pb.inc(1))
         .unwrap();
-    assert_eq!(sparse_vector_index_mmap.indexed_vector_count(), NUM_VECTORS);
+    pb.finish_and_clear();
+    assert_eq!(sparse_vector_index_mmap.indexed_vector_count(), vectors_len);
 
     // intent: bench `search` without filter on mmap inverted index
     group.bench_function("mmap-inverted-index-search", |b| {
         b.iter(|| {
             let results = sparse_vector_index_mmap
                 .search(
-                    &[&query_vector],
+                    &[&query_vector_it.next().unwrap().clone().into()],
```

Comment:
I have learned recently about [Bencher::iter_batched](https://docs.rs/criterion/latest/criterion/struct.Bencher.html#method.iter_batched] to offload the input generation to criterion.

Maybe it could help you here.